### ISSUE TITLE
Innosend

### ISSUE THREAD

**Issue by GoogleCodeExporter** [satyajit319/websmsdroid#25]
```
Totally love your app. Best Sms sender on the android platform.

Do you currently thing about including new providers? Innosend is a really
cheap one in Germany Sms start at 3 cent (without your own number) to 7.9
cent (own number). You also get 7 free sms per day!

There already is an android app supporting it GSchickt, which is ok but it
would be nice to have everything in one place, plus its a little buggy.

Thx a lot

Michael
```

Original issue reported on code.google.com by `mcsvo...@googlemail.com` on 9 Nov 2009 at 10:24

**Comment by GoogleCodeExporter** [comment]
```
the problem is, that i don't have an account to test it.
so. i'll think about it :)
```

Original comment by `felix.bechstein` on 9 Nov 2009 at 5:05
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
found the api of innosend:
http://www.innosend.de/downloads/innosend-api-sms.pdf
```

Original comment by `felix.bechstein` on 9 Nov 2009 at 7:26

**Comment by GoogleCodeExporter** [comment]
```
The setup of an account is free, you get a few cents to test (like 20 or so)
To charge the account there are two options
- bank account starting at 5 €
- paypal 10 €
If you have a dontion account I am sure there will be some people (like me) who 
will
be happy to help out with the basic costs

it's actually a nice serivce with sms starting at 0.03 cents (0.025 cents for 
slow
routing) 10 € last quiet a while!
```

Original comment by `mcsvo...@googlemail.com` on 9 Nov 2009 at 7:51

**Comment by GoogleCodeExporter** [comment]
```
Actually a problem might be the multible choices to send sms (gui vise)
- free sms
- with sender number (fast routing)
- without sender number (fast routing)
- with sender number (slow routing)
- without sender number (slow routing)

to my experience slow routing is pretty much useless since it is quiet slow and 
not
as reliable
```

Original comment by `mcsvo...@googlemail.com` on 9 Nov 2009 at 8:10

**Comment by GoogleCodeExporter** [comment]
```
the api works quite well. getting balance working allready.
sending sms and implementing paid sms should be not much work.

any idea how to handle free/paid sms switch?
i would prefer handle it as a separate provider for easy switching..
```

Original comment by `felix.bechstein` on 10 Nov 2009 at 1:18
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Hm good point, seperate provider is probably the fastest and easiest way to 
implement
- is it maybe possible to enable disable providers (like with a checkbox in the
preferences)? like with innosend you would get 5 new providers (3 if you only 
count
the once I would consider usefull since cheap routing is not reliable)
- can't see any other good way to implement it, I really like your gui so I 
would not
change to much about it since it is pretty streamlined 

There are some things that you might wanne keep in mind
- free sms is a little tricky but definatly a killer feature for the application
    - free sms are only 146 characters (maybe it is possible to let the counter count
down from 146 to 0 or stop the input at 146?
    - only 1 per hour (but I think it is possible to catch the error message from the api)
    - now you can get information about your remaining free sms at gmx and your left
money at sipgate by clicking the update icon, it would be nice to display   the 
money
left in your innosend account and the remaining time until the next free sms 
(both
information can be requested by the api)
```

Original comment by `mcsvo...@googlemail.com` on 10 Nov 2009 at 5:49

**Comment by GoogleCodeExporter** [comment]
```
added innosend support in trunk,
release coming very soon

please open new issues for further feature requests/bugs
```

Original comment by `felix.bechstein` on 12 Nov 2009 at 9:22
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
awesome thx a lot cant wait to test it
```

Original comment by `mcsvo...@googlemail.com` on 12 Nov 2009 at 9:50

**Comment by GoogleCodeExporter** [comment]
```
you may try this one: http://nossl.ub0r.de/~f/AndGMXsms-debug.apk
but don't forget to reinstall the app from market after release.
also uninstall/install is needed due to different certificates.
```

Original comment by `felix.bechstein` on 12 Nov 2009 at 9:57

**Comment by GoogleCodeExporter** [comment]
```
released v2.6 to market
```

Original comment by `felix.bechstein` on 13 Nov 2009 at 1:50
